test/types/schema.test.ts

```diff
@@ -737,3 +763,36 @@ function pluginOptions() {
   schema.plugin<any, SomePluginOptions>(pluginFunction2, { option2: 0 });
   expectError(schema.plugin<any, SomePluginOptions>(pluginFunction2, {})); // should error because "option2" is not optional
 }
+
+function gh12205() {
+  const campaignSchema = new Schema(
+    {
+      client: {
+        type: new Types.ObjectId(),
+        required: true
+      }
+    },
+    { timestamps: true }
+  );
+
+  const Campaign = model('Campaign', campaignSchema);
+  const doc = new Campaign();
+  expectType<Types.ObjectId>(doc.client);
+
+  type ICampaign = InferSchemaType<typeof campaignSchema>;
+  expectType<{ client: Types.ObjectId }>({} as ICampaign);
+
+  expectType<'type'>({} as ObtainSchemaGeneric<typeof campaignSchema, 'TPathTypeKey'>);
+
+  type A = ObtainDocumentType<{ client: { type: Schema.Types.ObjectId, required: true } }>;
```

Comment:
You use ```Schema.Types.ObjectId``` as type, I think this is incorrect.
Same thing for the following tests.